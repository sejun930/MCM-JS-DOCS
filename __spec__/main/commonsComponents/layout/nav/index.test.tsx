import { render } from "@testing-library/react";
import LayoutNavPage from "src/main/commonsComponents/layout/nav";

export default describe("Layout Nav Page", () => {
  // 스냅샷
  test("Snapshot", () => {
    const { container } = render(<LayoutNavPage />);
    expect(container).toMatchSnapshot();
  });

  // tap (a, p Tag)가 제대로 동작하는지 검증
  test("Snapshot", () => {
    const { container } = render(<LayoutNavPage />);

    const listEle = container.querySelector(".nav-list-wrapper");
    expect(listEle).toBeInTheDocument();

    // list가 존재하는지 검증
    if (listEle) {
      const aEle = listEle.querySelector(".tap");
      expect(aEle).toBeInTheDocument(); // a 태그가 존재하는지 검증

      if (aEle) {
        expect(aEle).toHaveAttribute("href", "/modules/modal");
        const pEle = aEle.querySelector(".tap-name");

        expect(pEle).toBeInTheDocument();
      }
    }
  });
});
