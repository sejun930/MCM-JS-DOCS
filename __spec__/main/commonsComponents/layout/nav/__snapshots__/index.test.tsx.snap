// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Nav Page Layout Nav Page - Snapshot 1`] = `
<div>
  <nav
    class="nav-wrapper css-g3duar"
  >
    <article
      class="nav-list-wrapper css-10y3vds"
    >
      <div
        class="nav-search-wrapper css-lobae1"
      >
        <form
          class="nav-search-form css-0"
        >
          <fieldset
            class="css-1opjr3t"
          >
            <legend>
              모듈 검색
            </legend>
            <div
              class="css-1rbly99"
            >
              <div
                class="css-1sg36r4"
              >
                123123
              </div>
            </div>
          </fieldset>
        </form>
      </div>
      <a
        class="mcm-link-unit tap css-1h5x3dy"
        href="/modules/modal"
      >
        <p
          class="mcm-p-unit tap-name css-pzlb5d"
        >
          Modal
        </p>
      </a>
    </article>
  </nav>
</div>
`;
